python/paddleserver/paddleserver/model.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 kubeflow.org.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+import os
+import numpy as np
+from paddle import inference
+import kfserving
+
+
+class PaddleModel(kfserving.KFModel):
+
+    def __init__(self, name: str, model_dir: str):
+        super().__init__(name)
+        self.name = name
+        self.model_dir = model_dir
+        self.ready = False
+        self.predictor = None
+        self.input_tensor = None
+        self.output_tensor = None
+
+    def load(self) -> bool:
+        def get_model_files(ext: str) -> str:
+            file_list = []
+            for filename in os.listdir(model_path):
+                if filename.endswith(ext):
+                    file_list.append(filename)
+            if len(file_list) == 0:
+                raise Exception("Missing {} model file".format(ext))
+            if len(file_list) > 1:
+                raise Exception("More than one {} model file".format(ext))
+            return os.path.join(model_path, file_list[0])
+
+        model_path = kfserving.Storage.download(self.model_dir)
+        config = inference.Config(get_model_files('.pdmodel'), get_model_files('.pdiparams'))
+        # TODO: add GPU support
+        config.disable_gpu()
+
+        self.predictor = inference.create_predictor(config)
+
+        # TODO: add support for multiple input_names/output_names
+        # are reusing these tensors multi-thread safe? or does it need to be?
```

Comment:
remember in multiprocessing mode the load is called before the process forking

Reply:
While multithread safety isn't guaranteed here, it seems multithreading isn't happening here, but multiprocessing, which shouldn't be a problem.